### PR TITLE
RP2-872 Update GET appointment endpoint

### PR DESCRIPTION
**Pull request by JamesBoobier**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/data/Appointments.kt [hunk 1]
```diff
@@ -47,8 +47,4 @@ data class Address(
 
 data class AppointmentsList(
   val results: List<Appointment> = listOf(),
-  val totalElements: Int,
```
Comment (JamesBoobier) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/data/Appointments.kt:50]: This is never used in the UI so safe to delete

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/AppointmentsService.kt [hunk 2]
```diff
@@ -100,6 +104,26 @@ class AppointmentsService(
     return appointmentList
   }
 
+  fun mapAppointmentsFromDatabase(deliusContacts: List<DeliusContactEntity>) = deliusContacts.map { deliusContact ->
+    val customFieldsFromNotes = getCustomFieldsFromNotes(deliusContact.notes, deliusContact.id)
+    Appointment(
+      title = extractSectionFromNotes(customFieldsFromNotes, APPOINTMENT_TITLE, deliusContact.id), // TODO put all sections in constants
```
Comment (tomhalson) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/AppointmentsService.kt:110]: is this TODO still needed? if so can we raise a tech debt ticket?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/ServiceUtils.kt [hunk 3]
```diff
@@ -75,3 +76,18 @@ fun getClaimFromJWTToken(token: String, claimName: String): String? {
   val jwtClaimsSet = JWTParser.parse(token.replaceFirst("Bearer ", "")).jwtClaimsSet
   return jwtClaimsSet.getStringClaim(claimName)
 }
+
+fun getCustomFieldsFromNotes(notes: String, id: Long?): List<String> {
+  try {
+    return notes.split(Regex("(^|\\n)###\\n"))[1].split(Regex("\\n")) // TODO put ### in constant
```
Comment (tomhalson) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/ServiceUtils.kt:82]: same question for this TODO?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/AppointmentsService.kt (hunk 2 above)
Comment (JamesBoobier) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/AppointmentsService.kt:110]: Fixed

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/ServiceUtils.kt (hunk 3 above)
Comment (JamesBoobier) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/ServiceUtils.kt:82]: Fixed

**Review by tomhalson — APPROVED** [review]

**Pull request merged**